mne/preprocessing/GradientRemover.py

```diff
@@ -1,7 +1,7 @@
-import mne
 import numpy as np
 from scipy.signal import detrend
 
+
 class GradientRemover:
```

Comment:
can we avoid a new and use a plain function?

Reply:
yes you can have helper/aux functions but if not present in the public API they should start with `_`

Reply:
Would it be useful to retain the feature where templates, detrending, and correction can be pulled for one TR only rather than the whole dataset, as part of the public API? Or should this correct a whole data set at once only?

Reply:
Maybe I'm missing the point here. Why do you need a state and therefore an object? Is it a two phase processing? you fit on some data and then you clean many datasets eg like for ICA?

Reply:
Actually, I forgot, there is one case where this might be done: if one wants real-time gradient correction, a template should be maintained and modifed, and then apply the correction one TR or sample at a time (this is not easily supported right now, but the object could be extended to do so, which is why I originally wrote it this way).

Reply:
An additional case would be if you wanted a non-template method (none are implemented here, but you could in theory use PCA or ICA to do the correction instead).